app/services/article_search_service.rb

```diff
@@ -0,0 +1,33 @@
+# Uses the Blacklight JSON API to search and then extracts select EDS fields
+class ArticleSearchService < AbstractSearchService
+  def initialize(options = {})
+    options[:query_url] ||= Settings.EDS_QUERY_URL.to_s
+    options[:response_class] ||= Response
+    super
+  end
+
+  class Response < AbstractSearchService::Response
+    def total
+      @json ||= JSON.parse(@body)
+      @json['response']['pages']['total_count'].to_i
+    end
+
+    def results
+      @json ||= JSON.parse(@body)
+      solr_docs = @json['response']['docs']
+      solr_docs.collect do |doc|
+        result = OpenStruct.new
+        result.title = doc['eds_title']
+        result.link = Settings.EDS_FETCH_URL.to_s % { id: doc['id'] }
+        result.id = doc['id']
+        result.description = doc['eds_abstract']
+        result
+      end
+    end
+
+    def facets
+      @json ||= JSON.parse(@body)
```

Comment:
What would you think of throwing the json parse into a private method?  I see that it's getting memoized into the `@json` ivar, so it's not really getting parsed multiple times.

I could see trying to avoid a liskov violation, but I think a private method is okay in this respect?